test/decorators_test.py

```diff
@@ -50,32 +70,654 @@ def param_definition():
     }
 
 
-@pytest.fixture(params=[True, False])
-def wrap_functions(request):
-    brewtils.decorators._wrap_functions = request.param
-    yield
-    brewtils.decorators._wrap_functions = False
+@pytest.fixture
+def param(param_definition):
+    return Parameter(**param_definition)
+
+
+@pytest.fixture
+def nested_1():
+    class NestedModel1(object):
+        parameters = [
+            Parameter(
+                key="key2",
+                type="String",
+                multi=False,
+                display_name="y",
+                optional=False,
+                default="100",
+                description="key2",
+            )
+        ]
+
+    return NestedModel1
+
+
+@pytest.fixture
+def nested_2():
+    class NestedModel2(object):
+        parameters = [
+            Parameter(
+                key="key3",
+                type="String",
+                multi=False,
+                display_name="z",
+                optional=False,
+                default="101",
+                description="key3",
+            )
+        ]
+
+    return NestedModel2
+
+
+class TestOverall(object):
+    """Test end-to-end functionality"""
+
+    def test_generate_params(self):
+        class Unused(object):
+            @command
+            def _cmd(self, x, y="some_default"):
+                return x, y
```

Comment:
Please add the following test scenarios:

```python
@command
def _cmd_bool(self, x=True)
    return x
assert param_x.type == "Boolean"

@command
def _cmd_float(self, x=1.0)
    return x
assert param_x.type == "Float"

@command
def _cmd_int(self, x=1)
    return x
assert param_x.type == "Integer"

@command
def _cmd_str(self, x="1")
    return x
assert param_x.type == "String"

@command
def _cmd_list_str(self, x=["1"])
    return x
assert param_x.type == "String"
assert param_x.multi = True


```

Reply:
These are all going to fail - in the test you're referencing we don't do any type assertions:

```python
param_y = cmds[0].get_parameter_by_key("y")
assert param_y.key == "y"
assert param_y.default == "some_default"
assert param_y.optional is True
```

That's because we don't currently make any assumptions with regards to parameter type. If you don't explicitly set the type then the type is "Any." I'll add assertions for that in the test that currently exists.

Like I mentioned on an earlier comment - I'm not opposed to adding a feature where we try to interpret types, but that's a new feature for sure, so I don't think this this PR is good place to do that. It may even require a major version bump. When we do add that functionality we will definitely need all of those tests (the last one in particular is going to be very ... fun).